### PR TITLE
[PIR-1523] - SQL queries for the prompts In Interactive reports are executed twice

### PR DESCRIPTION
**Pull request by miguelappleton**
@pentaho/tatooine_dev

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 8m 32s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
engine/extensions-pentaho-metadata
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.reporting.engine.classic.extensions.datasources.pmd.SimplePmdDataFactoryTest.testGetContextKeyParentIdentifier</b> (click to expand)</summary><p>

```
Parent given must not be null
```

</p></details>

  
Tests run: **12**, Failures: **1**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/reporting/job/PR-1677/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 8m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
engine/extensions-pentaho-metadata
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/reporting/job/PR-1677/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 3m 38s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
engine/extensions-pentaho-metadata
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/reporting/job/PR-1677/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by pentaho-aschurman** [comment]
@miguelappleton , My review... Great , thanks for adding the parameter. We need to add this parameter to the Self-Documentation as well please.

File: engine/extensions-pentaho-metadata/src/main/java/org/pentaho/reporting/engine/classic/extensions/datasources/pmd/SimplePmdDataFactory.java [hunk 1]
```diff
@@ -92,6 +94,7 @@ protected boolean isExpandArrays() {
   private String domainId;
   private String xmiFile;
   private IPmdConnectionProvider connectionProvider;
+  private boolean useParentIdentifier;
```
Comment (befc) [inline engine/extensions-pentaho-metadata/src/main/java/org/pentaho/reporting/engine/classic/extensions/datasources/pmd/SimplePmdDataFactory.java:97]: Do you think that declaring this boolean property default value would improve readability? 
```suggestion
  private boolean useParentIdentifier = false;
```

Comment (miguelappleton) [reply engine/extensions-pentaho-metadata/src/main/java/org/pentaho/reporting/engine/classic/extensions/datasources/pmd/SimplePmdDataFactory.java:97]: Yeah, perhaps. I don't see any problem, sounds good 👍

Comment (miguelappleton) [reply engine/extensions-pentaho-metadata/src/main/java/org/pentaho/reporting/engine/classic/extensions/datasources/pmd/SimplePmdDataFactory.java:97]: Perhaps. Ideally I would like to setup the variable with the value from config straight away, but I have to do it in the `initialize` function as I need to have the configurations loaded, therefore I cannot do it in the constructor (in a static block, for example). Either way, the 2 declarations above are equivalent, so I don't see any reason to oppose your suggestion.

Comment (miguelappleton) [reply engine/extensions-pentaho-metadata/src/main/java/org/pentaho/reporting/engine/classic/extensions/datasources/pmd/SimplePmdDataFactory.java]: @befc made the changes 👍

**Comment by buildguy** [comment]
### :x: Build failed in 54s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
engine/extensions-pentaho-metadata
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 21s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
engine/extensions-pentaho-metadata
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/reporting/job/PR-1677/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by befc — APPROVED** [review]

**Pull request merged**